File: JavaKotlinDers12RetrofitKriptoParaUygulamasi/app/src/main/java/com/example/javakotlinders12retrofitkriptoparauygulamasi/model/CryptoModel.java
package com.example.javakotlinders12retrofitkriptoparauygulamasi.model;

import com.google.gson.annotations.SerializedName;

public class CryptoModel {
    @SerializedName("currency") //ama burdaki isimler json çıktımdaki isimlerle aynı olmak zorunda
    String currency;//burdaki isim aynı olmak zorunda değil
    @SerializedName("price")
    String price;

    public String getCurrency() {
        return currency;
    }


    public String getPrice() {
        return price;
    }

}

File: KotlinDers13Retrofit/app/src/main/java/com/example/kotlinders13retrofit/model/CryptoModel.kt
package com.example.kotlinders13retrofit.model

import com.google.gson.annotations.SerializedName

//data snıfıları: içerisine tamamen veri çekeceğimiz ve verileri işleyeceğimiz yapı vardır.
data class CryptoModel(
    //@SerializedName("currency") //bi tane değer gelecek parametresi currency olucak.Onu al burdaki currency değişkenine ata demek
    val currency:String,   //bu değişkenin ismi önemli değil.Ama serializedde önemli aynısı olmalı json çıktısındaki ile
    //ama json çıktısı ile aynı isimdeyse zaten serialized kısmını yazmana gerek yok aslında
    //@SerializedName("price")
    val price:String) {


}